### PR TITLE
RestorationData

### PR DESCRIPTION
**Pull request by md0u80c9**
Amended ORKTaskViewController initWithTask to take a nullable restorationData field.
Updated ORKTest to allow us to test this out.
tested with NavigableOrderedTask.

### CODE REVIEW

File: ResearchKit/Common/ORKTaskViewController.m [hunk 1]
```diff
@@ -302,7 +302,7 @@ - (instancetype)initWithTask:(id<ORKTask>)task restorationData:(NSData *)data de
     
     self = [self initWithTask:task taskRunUUID:nil];
     
-    if (self) {
+    if (self && (data != nil)) {
         self.delegate = delegate;
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKTaskViewController.m]: Should accept `delegate` regardless of restoration. 
Shall we move `self.delegate = delegate;` out of `if block`?

**Comment by md0u80c9** [comment]
Agreed! Come to think of it - shouldn't there be a convenience initialiser for taskRunUUID which accepts delegate as a parameter too?

File: ResearchKit/Common/ORKTaskViewController.m [hunk 2]
```diff
@@ -304,10 +304,13 @@ - (instancetype)initWithTask:(id<ORKTask>)task restorationData:(NSData *)data de
     
     if (self) {
         self.delegate = delegate;
-        self.restorationClass = [self class];
-        NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc] initForReadingWithData:data];
-        [self decodeRestorableStateWithCoder:unarchiver];
-        [self applicationFinishedRestoringState];
+        if (data != nil) {
+            self.delegate = delegate;
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKTaskViewController.m]: Remove this extra assignment.

**Comment by YuanZhu-apple** [comment]
`delegate` can play a very important role in the restoration process (`applicationFinishedRestoringState`) during init. I think we can leave out `delegate` in the other initializer.

**Comment by md0u80c9** [comment]
Agreed.

**Comment by YuanZhu-apple** [comment]
@md0u80c9 Thanks! Merging it.

**Pull request merged**